File: app/src/main/java/de/karzek/diettracker/presentation/search/recipe/recipeEditDetails/adapter/viewHolder/RecipeEditDetailsAddViewHolder.java
package de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.viewHolder;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.Button;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import de.karzek.diettracker.R;
import de.karzek.diettracker.presentation.main.cookbook.recipeManipulation.adapter.itemWrapper.RecipeManipulationViewItemWrapper;
import de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.itemWrapper.RecipeEditDetailsViewItemWrapper;

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
public class RecipeEditDetailsAddViewHolder extends RecyclerView.ViewHolder {

    private final OnSaveRecipeClickedListener onSaveRecipeClickedListener;

    @BindView(R.id.add_recipe)
    Button addRecipe;

    public RecipeEditDetailsAddViewHolder(ViewGroup viewGroup,
                                          OnSaveRecipeClickedListener onSaveRecipeClickedListener) {
        super(LayoutInflater.from(viewGroup.getContext())
                .inflate(R.layout.viewholder_recipe_save, viewGroup, false));
        ButterKnife.bind(this, itemView);

        this.onSaveRecipeClickedListener = onSaveRecipeClickedListener;
    }

    public void bind(RecipeEditDetailsViewItemWrapper item) {
        addRecipe.setText(R.string.button_add);
    }

    @OnClick(R.id.add_recipe) public void onAddRecipeToDiaryClicked() {
        onSaveRecipeClickedListener.onSaveRecipeClicked();
    }

    public interface OnSaveRecipeClickedListener {
        void onSaveRecipeClicked();
    }

}

File: app/src/main/java/de/karzek/diettracker/presentation/search/recipe/recipeEditDetails/adapter/viewHolder/RecipeEditDetailsAddViewHolder.kt
package de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.viewHolder

import android.view.View
import androidx.recyclerview.widget.RecyclerView
import de.karzek.diettracker.R
import de.karzek.diettracker.presentation.search.recipe.recipeEditDetails.adapter.itemWrapper.RecipeEditDetailsViewItemWrapper
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.viewholder_recipe_save.*

/**
 * Created by MarjanaKarzek on 30.05.2018.
 *
 * @author Marjana Karzek
 * @version 1.0
 * @date 30.05.2018
 */
class RecipeEditDetailsAddViewHolder(override val containerView: View,
                                     private val onSaveRecipeClickedListener: OnSaveRecipeClickedListener) : RecyclerView.ViewHolder(containerView), LayoutContainer {

    fun bind(item: RecipeEditDetailsViewItemWrapper) {
        add_recipe.setText(R.string.button_add)
        add_recipe.setOnClickListener { onSaveRecipeClickedListener.onSaveRecipeClicked() }
    }

    interface OnSaveRecipeClickedListener {
        fun onSaveRecipeClicked()
    }

}
